src/core/raster/qgsrasterlayertemporalproperties.h

```diff
@@ -0,0 +1,116 @@
+/***************************************************************************
+                         qgsrasterlayertemporalproperties.h
+                         ---------------
+    begin                : February 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef QGSRASTERLAYERTEMPORALPROPERTIES_H
+#define QGSRASTERLAYERTEMPORALPROPERTIES_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsrange.h"
+#include "qgsmaplayertemporalproperties.h"
+#include "qgsrasterdataprovider.h"
+
+/**
+ * \class QgsRasterLayerTemporalProperties
+ * \ingroup core
+ * Implementation of map layer temporal properties for raster layers.
+ *
+ * \since QGIS 3.14
+ */
+
+class CORE_EXPORT QgsRasterLayerTemporalProperties : public QgsMapLayerTemporalProperties
+{
+  public:
+
+    /**
+     * Constructor for QgsRasterLayerTemporalProperties.
+     */
+    QgsRasterLayerTemporalProperties( bool enabled = false );
+
+    virtual ~QgsRasterLayerTemporalProperties() = default;
+
+    /**
+     * Mode of the raster temporal properties
+     *
+     **/
+    enum TemporalMode
+    {
+      ModeFixedTemporalRange, //! Mode when temporal properties have fixed start and end datetimes.
+      ModeTemporalRangeFromDataProvider, //! Mode when raster layer depends on temporal range from its data provider.
+      ModeTemporalRangesList //! To be used when raster layer has list of temporal ranges.
```

Comment:
As far as I can see there's no api for handling this mode yet -- if not, better comment it out or remove it to avoid confusion